tests/translator/input/api_with_usageplans.yaml

```diff
@@ -0,0 +1,152 @@
+Resources:
+   MyApiOne:
+     Type: AWS::Serverless::Api
+     Properties:
+       StageName: Prod
+       Auth:
+         ApiKeyRequired: true
+         UsagePlan:
+          CreateUsagePlan: SHARED
+
+   MyApiTwo:
+     Type: AWS::Serverless::Api
+     Properties:
+       StageName: Prod
+       Auth:
+         ApiKeyRequired: true
+         UsagePlan:
+          CreateUsagePlan: NONE
+          UsagePlanId: !Ref MyApiTwoUsagePlan
```

Comment:
My thought with this-

I thought that, if `CreateUsagePlan: None` is set, we won't create the `UsagePlanKey` or `ApiKey`. Or, did we decide to create those if the `UsagePlanId` is set, and if that property isn't set, we don't do anything?

I want to make sure for the case where, if someone has set up their UsagePlan, ApiKey, and UsagePlanKey, we don't mess things up and they can turn off the auto-generated SAM portions.

Reply:
I have documented this [here](https://github.com/awslabs/serverless-application-model/pull/1179/files#diff-ebb9eb355c6f4c24d565c7e8517549a5R920). 

When `CreateUsagePlan:NONE` it works similar to `CreateusagePlan:SINGLE` where we create `ApiKey and UsagePlanApiKey` and expect the `UsagePlan` to be defined by the user. If the user has already set up the three resources (they don't want to create a usage plan) they will not add `UsagePlan` section for that `Api`. One problem here would be if `CreateUsagePlan` is added in the globals section we cannot turn it off for a single Api at the `Resources` level. The user has to either give a `UsagePlanId` or allow us to create the `UsagePlan`. The workaround for this would be to add `CreateUsagePlan` for the required Apis at the `Resource` level instead of adding in the `Globals` section.
Currently, this code throws error when `CreateUsagePlan:None` is given and `usagePlanId` is not given. Instead of throwing error I could do:
1) create the `ApiKey`, `UsagePlanApiKey` when `UsagePlanId` is given
2) do not create any resource when `UsagePlanId` is not given